pages/api/v1/subscanApi/index.tsx

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import { NextApiHandler } from 'next';
+import storeApiKeyUsage from '~src/api-middlewares/storeApiKeyUsage';
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import { getCache, setCache } from '~src/api-utils/subscanCachedData';
+import { generateKey } from '~src/util/getRedisKeys';
+
+export const SUBSCAN_API_KEY = '74d1845ab15f4b889a64dfef074ef222';
+
+export const SUBSCAN_API_HEADERS = {
+	Accept: 'application/json',
+	'Content-Type': 'application/json',
+	'X-API-Key': SUBSCAN_API_KEY
+};
+
+export const getSubscanData = async (url: string, body?: any, headers?: any, method?: string) => {
+	try {
+		const data = await (
+			await fetch(url, {
+				body: body ? JSON.stringify(body) : '',
+				headers: headers || SUBSCAN_API_HEADERS,
+				method: method || 'POST'
+			})
+		).json();
+
+		return data;
+	} catch (error) {
+		return error;
+	}
+};
+
+const handler: NextApiHandler<{ data: any } | { error: string | null }> = async (req, res) => {
+	storeApiKeyUsage(req);
+
+	const { keys, url, body, method, headers, cacheEnabled = true, hardReload = false, expiry } = req.body;
```

Comment:
keys should be set in BE, else anyone can set any key's value using the api
headers and expiry should be set in BE too

Reply:
cacheEnabled should be read from env, with the same value as it is used for other redis caches